colossalai/auto_parallel/passes/meta_info_prop.py

```diff
@@ -0,0 +1,158 @@
+        graph_info.fwd_in = output_tensors
+        node.meta = {**asdict(graph_info)}
+
+    @compatibility(is_backward_compatible=False)
+    def node_handler(self, node: Node) -> None:
+        """
+        Handle other kind of nodes
+        """
+        assert hasattr(node, 'best_metainfo'), f"Cannot find best_metainfo in node {node}"
+        graph_info = GraphInfo()
+        meta_info = node.best_metainfo
+        meta_info: MetaInfo
+
+        # set data_ptr for input_tensor in MetaInfo class
+        input_tensor: List[torch.Tensor] = meta_info.fwd_in
+        buffer_tensor: List[torch.Tensor] = meta_info.fwd_buffer
+        output_tensor: List[torch.Tensor] = meta_info.fwd_out
+
+        if len(input_tensor) > 0:
+            for par in node._input_nodes:
+                if par.meta:
+                    if len(par.meta["fwd_out"]) > 0:
+                        # set data_ptr for the input_tensor of current node from the output_tensor of its parent node
+                        for tensor in par.meta["fwd_out"]:
+                            tensor: torch.Tensor
+                            target_tensor = next(
+                                (x for x in input_tensor if not x.data_ptr() and x.shape == tensor.shape), None)
+                            target_tensor.data_ptr = tensor.data_ptr
+
+            # set data_ptr for tensor in input_tensor that is not set
+            for tensor in input_tensor:
+                if not tensor.data_ptr():
+                    self._set_data_ptr(tensor)
+
+        # attach it to graph_info
+        graph_info.fwd_in = input_tensor
+
+        if self._is_inpalce(node):
+            # inplace operation will not create new tensor
+            # set data_ptr for buffer_tensor and output_tensor of current node
+            for tensor in input_tensor:
+                tensor: torch.Tensor
+                target_buffer_tensor = next((x for x in buffer_tensor if not x.data_ptr() and x.shape == tensor.shape),
+                                            None)
+                target_output_tensor = next((x for x in output_tensor if not x.data_ptr() and x.shape == tensor.shape),
+                                            None)
+                target_buffer_tensor.data_ptr = tensor.data_ptr
+                target_output_tensor.data_ptr = tensor.data_ptr
+            # attach them to graph_info
+            graph_info.fwd_tmp = buffer_tensor
+            graph_info.fwd_out = output_tensor
+
+        else:
+            # set data_ptr for buffer_tensor
+            for tensor in buffer_tensor:
+                self._set_data_ptr(tensor)
+            # attach it to graph_info
+            graph_info.fwd_tmp = buffer_tensor
+
+            # set data_ptr for output_tensor
+            for tensor in output_tensor:
+                self._set_data_ptr(tensor)
+            # attach it to graph_info
+            graph_info.fwd_out = output_tensor
+
+        # fetch other memory informations
```

Comment:
@super-dainiu I fetch temp memory informations here and store them in `GraphInfo.fwd_mem_tmp` and `GraphInfo.bwd_mem_tmp`